### ISSUE TITLE
Cache vcpkg packages

### ISSUE THREAD

**Issue by JamesYangLim** [lukka/CppBuildTasks#11]
Hi @lukka, I have been using CppBuildTasks for awhile and it has been great. Just have a few question here with pipeline caching feature.

**Question 1:**
Your example is caching ```'$(Build.BinariesDirectory)/vcpkg'```, the entire folder of vcpkg. For my case, the whole vcpkg folder is getting up to around 5GB which is big to cache. I wonder if you have an example to cache ```'$(Build.BinariesDirectory)/vcpkg/installed'``` folder?

**Question 2:**
After restored the cache, run vcpkg task will remove the outdated packages and reinstall them. This all go well except the cache task does not recache the updated packages because the cache key isn't invalidated. The only solution I can think of at the moment is to use a new cache key, invalidate the existing cache key and redo all vcpkg task from scratch which is not nice.

Let say I want to install one more package and update two packages, I would wish that the cache task will restore the installed packages, vcpkg remove outdated packages and reinstall the new version, vcpkg install the new added package, and finally recache the whole installed packages in post-job. This will avoid to redo vcpkg task from scratch if there is new package version or just adding one new package.

Any thoughts?

**Comment by JamesYangLim** [comment]
I just saw this

```
Note: since buildtrees, packages and download subdirectories must not be cached, they are excluded by the .artifactignore file generated by the vcpkg-run task.
```

This answer my first question.

**Comment by JamesYangLim** [comment]
Hi @lukka, although .artifactignore excluded some folders, caching in my pipeline still go up to 5GB. Meanwhile installed folder is only roughly 750MB which is a lot smaller. So I think that it will be still be very useful if you have that example done. Thank you!

**Comment by lukka** [comment]
@JamesYangLim there is space for improvements indeed. 
In order to get the content and size of the first level of the cached content, could you run on server the following command just after the cache has been restored:

```
cd /path/to/vcpkg/root/dir
du -hs *
```

Also try to change the key's value given to the Cache task, and verify whether the size of the cache is going to be significantly smaller: this could happen if the cache task has been run with no `.artifactignore` file in the past (i.e. the cached content is immutable, and if cached once, it wont change unless a different key is being used)

**Comment by lukka** [comment]
@JamesYangLim found the issue and fixed it in #15  ! Thanks for reporting this problem.
